msgpack/src/object.cpp

```diff
@@ -0,0 +1,81 @@
+#include <object.hpp>
+#include <adaptor.hpp>
+
+CObject get_cobject(const Value &ark_object, ValueType type)
+{
+	CObject object;
+
+	if(type == ValueType::NFT)
+	{
+		if(ark_object == Ark::True)
+			object = true;
+		else if(ark_object == Ark::False)
+			object = false;
+	}
+	else if(type == ValueType::Number)
+	{
+		object = static_cast<Value>(ark_object).number();
+	}
+	else if(type == ValueType::String)
+	{
+		object = static_cast<Value>(ark_object).string_ref();
+	}
+	else
+	{	
+		object = static_cast<Value>(ark_object).list();
+	}
+	return object;
+}
+
+namespace ArkMsgpack
+{
+	// msgpack::sbuffer
+	Value msgpack_sbuffer(std::vector<Value> &args)
```

Comment:
the interface for functions is now `Value name(std::vector<Value>& args, Ark::VM* vm)`

Reply:
Up